components/automate-deployment/pkg/backup/builder_operations.go

```diff
@@ -0,0 +1,253 @@
+package backup
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type BuilderMinioDumpOperation struct {
+	Name       string   `json:"name"`
+	ObjectName []string `json:"storage_key"`
+}
+
+var _ Operation = &BuilderMinioDumpOperation{}
+
+func (d *BuilderMinioDumpOperation) Backup(backupCtx Context, om ObjectManifest, progChan chan OperationProgress) error {
+	backupBucketPrefix := d.backupBucketPrefix()
+	ctx := backupCtx.ctx
+
+	logrus.WithFields(logrus.Fields{
+		"name":      d.Name,
+		"backup_id": backupCtx.backupTask.TaskID(),
+		"prefix":    backupBucketPrefix,
+		"operation": "builder_minio_dump",
+		"action":    "backup",
+	}).Info("Running backup operation")
+
+	objects, _, err := backupCtx.builderBucket.List(ctx, "", false)
+	if err != nil {
+		return err
+	}
+	builderArtifacts, err := ioutil.TempFile("", "builderartifacts")
+	if err != nil {
+		return err
+	}
+	defer builderArtifacts.Close()
+	defer os.Remove(builderArtifacts.Name())
```

Comment:
i dont think this needs to be defered

Reply:
I assume we are writing to a file here (as apposed to a byte.Buffer) because we are worried about keeping the full manifest in memory?

Reply:
we could probably replace the with a reader of some sort, but i guess its fine for now